platform/common/core/generic/vault/interceptor.go

```diff
@@ -365,6 +368,8 @@ func (i *Interceptor[V]) Reopen(qe VersionedQueryExecutor) error {
 }
 
 func (i *Interceptor[V]) IsClosed() bool {
+	i.Mutex.Lock()
```

Comment:
please, let's use a read lock here. We need also a lock when we set Closed to false.